aces/analysis/tp_baseline_sub.py

```diff
@@ -0,0 +1,308 @@
+
+    Returns:
+        dict: Contains spectrum array, spectral axis, validity mask, and index mappings
+             between masked and original arrays.
+    """
+    cube = spectral_cube.SpectralCube.read(file_path).to(u.K)
+    spectrum = cube.mean(axis=(1, 2))
+    spectral_axis = cube.spectral_axis.to("MHz")
+    spectrum_arr = np.array(spectrum)
+    valid_mask = ~np.isnan(spectrum_arr)
+
+    return {
+        "spectrum": spectrum_arr[valid_mask],
+        "spectral_axis": spectral_axis[valid_mask],
+        "mask": valid_mask,
+        "original_to_masked": np.cumsum(valid_mask) - 1,
+        "masked_to_original": np.arange(len(spectrum_arr))[valid_mask],
+    }
+
+
+def get_masked_ranges(line_free_ranges, mask, original_to_masked):
+    """
+    Convert spectral ranges to their masked array equivalents.
+
+    Args:
+        line_free_ranges: List of tuples containing start and end indices of line-free regions.
+        mask: Boolean array indicating valid data points.
+        original_to_masked: Array mapping original indices to masked array indices.
+
+    Returns:
+        list: Tuples of start and end indices in the masked array space.
+    """
+    masked_ranges = []
+    for start, end in line_free_ranges:
+        if start < len(mask) and mask[start]:
+            masked_start = original_to_masked[start]
+            end_channel = min(end, len(mask) - 1)
+            while end_channel > start and not mask[end_channel]:
+                end_channel -= 1
+            if end_channel > start:
+                masked_ranges.append((masked_start, original_to_masked[end_channel]))
+    return masked_ranges
+
+
+def auto_select_line_free_ranges_sigma_clip(spectrum_arr, min_range_length=10, sigma_threshold=2.0, max_iter=10):
+    """
+    Automatically identify line-free regions using iterative sigma clipping.
+
+    Args:
+        spectrum_arr: Array of spectral data.
+        min_range_length: Minimum length of a valid line-free range.
+        sigma_threshold: Number of standard deviations for clipping.
+        max_iter: Maximum number of sigma-clipping iterations.
+
+    Returns:
+        list: Tuples of start and end indices for line-free regions.
+    """
+    mask = np.ones_like(spectrum_arr, dtype=bool)
+    previous_mask = np.zeros_like(spectrum_arr, dtype=bool)
+    iteration = 0
+
+    while iteration < max_iter and not np.array_equal(mask, previous_mask):
+        previous_mask = mask.copy()
+        median_val = np.median(spectrum_arr[mask])
+        mad = np.median(np.abs(spectrum_arr[mask] - median_val))
+        sigma = 1.4826 * mad
+        mask = spectrum_arr < (median_val + sigma_threshold * sigma)
+        iteration += 1
+
+    ranges = []
+    i = 0
+    while i < len(mask):
+        if mask[i]:
+            start = i
+            while i < len(mask) and mask[i]:
+                i += 1
+            if (i - start) >= min_range_length:
+                ranges.append((start, i - 1))
+        else:
+            i += 1
```

Comment:
this is the same as `get_masked_ranges`, right?  I think this could be simplified to
```python
labels = scipy.ndimage.label(mask)[0]
slices = scipy.ndimage.find_objects(labels)
ranges = [(slc.start, slc.end) for slc in slices]
```

Reply:
I'm going to resolve this for now to permit merging, but I haven't addressed this yet. Added a TODO ...